src/persistentReducedStream.ts

```diff
@@ -14,26 +13,68 @@ import { defaultErrorSubject } from './internal/defaultErrorSubject';
 import { markName } from './internal/markers';
 import { tag } from 'rxjs-spy/operators';
 
-/**
- * A hot, persistent observable with a state field
- */
-export class StateStream<State> extends Observable<State>
-  implements SubscriptionLike {
+export class PersistentReducedStateStream<State> extends Observable<State> {
   private subject: BehaviorSubject<State>;
-  private subscription: Subscription;
-
-  constructor(initialState: State, reducedState$: Observable<State>) {
-    super();
-    this.subject = new BehaviorSubject(initialState);
+  private reducerSubscription?: Subscription;
+  public name: string;
+  private reducers: RegisteredReducer<State, any>[];
+  private errorSubject: Subject<any>;
 
+  /**
+   * Subscribes to the action stream and starts reducing this state stream.
+   * startReducing does not always immediately emit a value since the value
+   * might have already been emitted when the state stream was created.
+   *
+   * initialState can be passed to reset the state of the stream to some other value.
+   * This can be useful for hot-reloading
+   *
+   * startReducing will recreate the inner subject if it was previously closed
+   */
+  startReducing = (action$: ActionStream, initialState?: State) => {
+    // Reset the subject value if it doesn't match the desired initialState
+    // This is normally used for hot-reloading
+    if (this.subject.closed && initialState) {
+      this.subject = new BehaviorSubject(initialState);
+    } else if (initialState && this.subject.value !== initialState) {
+      this.subject.next(initialState);
+    }
     // NOTE: .subscribe(this.subject) does not work
     // All tests pass with it, but in practice there are cases
     // where the subject isn't updated as expected
-    this.subscription = reducedState$.subscribe({
-      complete: () => this.subject.complete(),
-      error: (err) => this.subject.error(err),
-      next: (state) => this.subject.next(state),
-    });
+    this.reducerSubscription = action$
+      .pipe(
+        combineReducers(initialState || this.subject.value, this.reducers, {
+          errorSubject: this.errorSubject,
+        }),
+        markName(this.name),
+        tag(this.name)
+      )
+      .subscribe({
+        complete: () => this.subject.complete(),
+        error: (err) => this.subject.error(err),
+        next: (state) => this.subject.next(state),
+      });
+    return this;
+  };
+
+  stopReducing = () => {
+    this.subject.unsubscribe();
```

Comment:
Why this and what is it doing? Is it needed?

Reply:
Yes, we must unsubscribe from the inner subject to close it. You can read a bit more about closing subjects here: https://ncjamieson.com/closed-subjects/

In practice, I believe this is not strictly necessary, so we could also avoid it, but then we would also need to remove all the logic related to checking whether the inner subject is closed - we could instead check whether the `reducerSubscription` has been set up.

Furthermore, if we don't close the inner subject, reading the latest state of a "closed" reducer stream would still be possible, and in my opinion this shouldn't be allowed since it could lead to reading outdated state.

Reply:
Ok, makes sense. I think it would also make sense to add some of this info as a comment.  The key from that linked article in this context is IMO
> Well, the unsubscribe method in the Subject class doesn’t actually unsubscribe anything. Instead, it marks the subject as closed and sets its internal array subscribed observers — Subject extends Observable, remember — to null.

Didn't knew that.